mobile/lib/features/wallet/payment_claimed_change_notifier.dart

```diff
@@ -1,20 +1,124 @@
+  late final SwapTradeValuesService tradeValuesService;
+
+  late final SwapTradeValues _sellTradeValues;
 
   void waitForPayment() => _claimed = false;
 
   bool isClaimed() => _claimed;
 
+  PaymentClaimedChangeNotifier(this.tradeValuesService) {
+    _sellTradeValues = _initDummyOrder();
+  }
+
+  SwapTradeValues _initDummyOrder() {
+    return SwapTradeValues.fromQuantity(
+        quantity: Amount(10),
+        leverage: Leverage(1),
+        price: null,
+        fundingRate: fundingRateSell,
+        direction: Direction.short,
+        tradeValuesService: tradeValuesService);
+  }
+
   @override
-  void notify(Event event) {
+  void notify(bridge.Event event) {
     if (event is bridge.Event_PaymentClaimed) {
-      _claimed = true;
-      super.notifyListeners();
+      final paymentAmountMsats = event.field0;
+      final paymentHash = event.field1;
+      final paymentAmountSats = paymentAmountMsats / 1000;
+
+      logger.i("Amount : $paymentAmountSats hash: $paymentHash");
+
+      if (rust.api.isUsdpPayment(paymentHash: paymentHash)) {
+        // TODO: how do I do this properly? According to stackoverflow one should not open a dialog from within a change notifier
+        logger.i("Received payment which should be converted to USDP");
+        var context = shellNavigatorKey.currentContext!;
+        final submitOrderChangeNotifier = context.read<SubmitOrderChangeNotifier>();
+        final margin = Amount(paymentAmountSats.ceil());
+
+        if (_sellTradeValues.price == null) {
+          logger.e(
+              "Could not convert received payments directly into USDP because we do not have a price");
+          _claimed = true;
+          final messenger = ScaffoldMessenger.of(context);
+          showSnackBar(messenger,
+              "Could not convert received payment into USDP. Please try todo it manually.");
+          notifyListeners();
+          return;
+        }
+
+        final price = _sellTradeValues.price!;
+        var fee = _sellTradeValues.fee;
+
+        final quantity = Amount(((paymentAmountSats / 100000000) * price).ceil());
+        logger.i("Posting order for quantity $quantity");
+
+        // TODO: why do we need these values when submitting a new order? :/
+        var liquidationPrice = 100000.0;
+        var fundingRate = 100000.0;
+        var expiry = DateTime.now();
+
+        submitOrderChangeNotifier.submitPendingOrder(
+            TradeValues(
+                direction: Direction.short,
+                margin: margin,
+                quantity: quantity,
+                leverage: Leverage(1.0),
+                price: price,
+                liquidationPrice: liquidationPrice,
+                fee: fee,
+                fundingRate: fundingRate,
+                expiry: expiry,
+                tradeValuesService: tradeValuesService),
+            PositionAction.open,
+            stable: true);
+
+        // TODO: figure this one out how to go back to home screen
```

Comment:
❓ in what context do you want to go back to the Home Screen?

Reply:
Oh , that's already resolved